analyzers/src/SonarAnalyzer.CSharp/Rules/AspNet/CallModelStateIsValid.cs

```diff
@@ -90,6 +87,11 @@ private static void ProcessCodeBlock(SonarCodeBlockStartAnalysisContext<SyntaxKi
         }
     }
 
+    private static bool HasApiControllerAttribute(ITypeSymbol type) =>
+        type is not null
```

Comment:
Why are you checking for null on the type?
Since you call `DerivesFrom` on the condition above, at line 53, you don't need to do it here.